src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -276,13 +291,13 @@ public function createAction(Request $request)
      * @return RedirectResponse
      */
     #[AdminSecurity("is_granted('read', request.get('_legacy_controller'))", redirectRoute: 'admin_orders_index')]
-    public function searchAction(Request $request)
-    {
-        /** @var ResponseBuilder $responseBuilder */
-        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
-
+    public function searchAction(
+        Request $request,
+        ResponseBuilder $responseBuilder,
+        #[Autowire(service: 'prestashop.core.grid.definition.factory.order')] OrderGridDefinitionFactory $orderGridDefinitionFactory,
+    ) {
```

Comment:
There is a helper from `PrestaShopAdminController` that you can use here `$this->buildSearchResponse` so you don't need to inject `ResponseBuilder`